src/enzyme_ad/jax/Passes/EnzymeHLOOpt.cpp

```diff
@@ -19672,6 +19672,54 @@ struct TransposeIf final
     return success();
   }
 };
+struct IfOpLiftCommonOps final
+    : public CheckedOpRewritePattern<stablehlo::IfOp, IfOpLiftCommonOps> {
+  using CheckedOpRewritePattern::CheckedOpRewritePattern;
+
+  LogicalResult matchAndRewriteImpl(stablehlo::IfOp op,
+                                    PatternRewriter &rewriter) const {
+    mlir::Region &trueRegion = op.getTrueBranch();
+    mlir::Region &falseRegion = op.getFalseBranch();
+
+    SmallVector<std::pair<Operation *, Operation *>> opsToLift;
+
+    auto &trueBlock = trueRegion.front();
+    auto &falseBlock = falseRegion.front();
+
+    for (auto trueIt = trueBlock.begin();
+         !trueIt->hasTrait<mlir::OpTrait::IsTerminator>(); ++trueIt) {
+      Operation *trueOp = &*trueIt;
+
+      for (auto falseIt = falseBlock.begin();
+           !falseIt->hasTrait<mlir::OpTrait::IsTerminator>(); ++falseIt) {
+        Operation *falseOp = &*falseIt;
+
+        if (OperationEquivalence::isEquivalentTo(
+                trueOp, falseOp, OperationEquivalence::IgnoreLocations)) {
+          opsToLift.emplace_back(trueOp, falseOp);
+          break;
+        }
+      }
+    }
+
+    if (opsToLift.empty())
+      return rewriter.notifyMatchFailure(op, "no common ops found");
+
+    for (auto [trueOp, falseOp] : opsToLift) {
+      Operation *liftedOp = rewriter.clone(*trueOp);
```

Comment:
instead could we perhaps do (assuming the relevant ordering)
```
rewriter.modifyOpInPlace(trueOp, [&](){
    trueOp->moveBefore(ifOp);
})
```

Reply:
done